src/nft/hooks/useFetchAssets.ts

```diff
@@ -100,3 +106,48 @@ export function useFetchAssets(): () => Promise<void> {
     tokenTradeInput,
   ])
 }
+
+export const useFetchSingleAsset = () => {
```

Comment:
A lot of this code is duplicated from `useFetchAssets`. Have you tried when a users hits the buy button to add it to the bag, keep the bag hidden, and then just call `useFetchAssets()`? This would pair well with the updated designs for handling price changes as then you'd just have to reveal the bag.

Reply:
The follow up would be if a user already has items in their bag. In that case you could add an optional `asset?` param to `useFetchAsset` which would override using the assets currently in the bag.

Reply:
This function would still need to touch bag state in the case of a price change since we'll need to display that change in the bag to the user to review. If we leverage `useFetchAssets` that logic is already handled for us. The only substantial change for the fetch query would be a conditional on `nftTrades: singleAsset ? buildNftTradeInput([singleAsset]) : buildNftTradeInputFromBagItems(itemsInBag),` the rest of the logic should be able to be left untouched.